### PR TITLE
Add ErrorMessage constructor that calls StrFormat()

### PR DESCRIPTION
**Pull request by dimitry-**
Test: builds

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/Result.h [hunk 1]
```diff
@@ -13,7 +15,10 @@
 class ErrorMessage final {
  public:
   ErrorMessage() = default;
-  explicit ErrorMessage(std::string message) : message_(std::move(message)) {}
+  explicit ErrorMessage(std::string message) : message_{std::move(message)} {}
+  template <typename... Args>
+  explicit constexpr ErrorMessage(const absl::FormatSpec<Args...>& format, const Args&... args)
+      : message_{absl::StrFormat(format, args...)} {}
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/Result.h:21]: Is there a chance that this new constructor is considered over the `std::string`-constructor when converting from something else? Maybe we should require at least 2 arguments for this constructor (format string and one arg) to be on the safe side, WDYT?

**Review by beckerhe — APPROVED** [review]
Very nice. Thanks. Otherwise LGTM.

File: src/OrbitBase/include/OrbitBase/Result.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/Result.h:21]: it does not seem to help, gcc is still unhappy.

Comment (beckerhe) [reply src/OrbitBase/include/OrbitBase/Result.h:21]: I don't see the problem directly either. I will have a look.

**Comment by dimitry-** [comment]
well, that apparently works only for clang.. I am going to close this for now. Not sure if there is a solution that would satisfy gcc and msvc

**Comment by beckerhe** [comment]
I'm sorry I haven't had time to really look into this. I believe there might be some SFINAE missing. But I need to understand how abseil handles FormatSpec.

**Pull request closed without merging**